narwhals/_compliant/when_then.py

```diff
@@ -154,11 +154,18 @@ def __call__(self, df: EagerDataFrameT, /) -> Sequence[EagerSeriesT]:
         is_expr = self._condition._is_expr
         when: EagerSeriesT = self._condition(df)[0]
         then: EagerSeriesT
+        if (
+            self._condition._metadata is not None
+            and self._condition._metadata.is_scalar_like
+        ):
+            when._broadcast = True
```

Comment:
i don't think we should ever broadcast `when`:

- either `when` preserves length, in which case we don't need to do anything
- or `when` is scalar-like, in which case `then` and `otherwise` must also be scalar-like

It's true that Polars supports `pl.when(when_scalar).then(then_column)`, but I don't think we should, just like how we don't support `df.filter(predicate_scalar)`